elisp/gen-metadata.el

```diff
@@ -0,0 +1,59 @@
+;;; gen-metadata.el --- generate package info file     -*- lexical-binding: t; -*-
+
+;; Copyright 2021 Google LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;     https://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+
+;;; Commentary:
+
+;; Generate a package metadata file.
+;;
+;; Usage:
+;;
+;;   emacs --quick --batch --load=gen-metadata.el \
+;;     --funcall=elisp/gen-metadata-and-exit SOURCE DEST
+;;
+;; Generates a metadata file DEST using the SOURCE -pkg.el.
+;; This is a JSON file that contains package metadata to allow easier extraction
+;; for non-Emacs tools.
+;;
+;; Exits with a zero status only if successful.
+
+;;; Code:
+
+(require 'package)
+(require 'json)
+
+(defun elisp/gen-metadata-and-exit ()
+  "Generate package metadata file and exit Emacs.
+See the file commentary for details."
+  (unless noninteractive
+    (error "This function works only in batch mode"))
+  (let* ((src (pop command-line-args-left))
```

Comment:
Consider using a `pcase` construct here similar to compile.el, and provide an error if the argument list is incorrect, in case someone changes the `.bzl` file and forgets to update this file.

Reply:
Done